lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -120,8 +121,15 @@ def load(cls, databag: MutableMapping):
             log.error(msg, exc_info=True)
             raise DataValidationError(msg) from e
 
-    def dump(self, databag: Optional[MutableMapping] = None):
-        """Write the contents of this model to Juju databag."""
+    def dump(self, databag: Optional[MutableMapping] = None, clear: bool = True):
+        """Write the contents of this model to Juju databag.
+
+        :param databag: the databag to write the data to.
+        :param clear: ensure the databag is cleared before writing it.
+        """
+        if clear and databag:
+            databag.clear()
```

Comment:
(In a future PR) we should rethink the `clear`. It seems un-jujuy to do so. 
- ops itself prevents assignment of relation data
- relation mgmt libs is our own invention; nothing prevents charm code or other libs to write to the same relation data

Reply:
`.defer()` --> https://github.com/canonical/traefik-k8s-operator/issues/252